apis/kubedb/v1alpha2/zookeeper_webhook.go

```diff
@@ -159,18 +160,41 @@ var zookeeperReservedVolumeMountPaths = []string{
 	ZooKeeperDataVolumePath,
 }
 
-func (z *ZooKeeper) validateVolumesMountPaths() error {
-	if z.Spec.PodTemplate.Spec.VolumeMounts == nil {
+func (z *ZooKeeper) validateVolumesMountPaths(podTemplate *ofst.PodTemplateSpec) error {
+	if podTemplate == nil {
 		return nil
 	}
-	rPaths := zookeeperReservedVolumeMountPaths
-	volumeMountPaths := z.Spec.PodTemplate.Spec.VolumeMounts
-	for _, rvm := range rPaths {
-		for _, ugv := range volumeMountPaths {
-			if ugv.Name == rvm {
-				return errors.New("Cannot use a reserve volume name: " + rvm)
+	if podTemplate.Spec.Containers == nil {
+		return nil
+	}
+
+	for _, rvmp := range zookeeperReservedVolumeMountPaths {
+		containerList := podTemplate.Spec.Containers
+		for i := range containerList {
+			mountPathList := containerList[i].VolumeMounts
+			for j := range mountPathList {
+				if mountPathList[j].MountPath == rvmp {
+					return errors.New("Can't use a reserve volume mount path name: " + rvmp)
+				}
 			}
 		}
 	}
+
+	if podTemplate.Spec.InitContainers == nil {
+		return nil
+	}
+
+	for _, rvmp := range zookeeperReservedVolumes {
```

Comment:
zookeeperReservedVolumeMountPaths

Reply:
As podTemplate.spec.containers & podTemplate.spec.initContainers are of same signature, We can even combine them together for this checking here

Reply:
Combined